docs/metasploit-framework.wiki/ad-certificates/overview.md

```diff
@@ -1,15 +1,119 @@
 ## What is AD CS?
 
-TODO
+Active Directory Certificate Services, also known as AD CS, is a Active Directory tool for
+letting administrators issue and manage public key certificates that can be used to
+connect to various services and principals on the domain. It is often used to provide
+certificates that can be used in place of credentials for logging into a network, or to
+provide certificates that can be used to sign and verify the authenticity of data.
 
-ADCS requires the following ports:
+The main guarantees that AD CS aims to provide are:
+- Confidentiality via encryption
+- Integrity via digital signatures
+- Authentication by associating certificate keys with computers, users, or device accounts
+  on a computer network.
 
-TODO
+Given that AD CS often holds highly sensitive keys and access credentials for a corporate
+network, this makes it a prime target for attackers.
+
+## Required Ports for AD CS
+Active Directory requires the following TCP ports be open on all domain controllers (taken from
+https://www.encryptionconsulting.com/ports-required-for-active-directory-and-pki/), which
+heavily overlaps with the ports required for AD CS (see https://learn.microsoft.com/en-us/archive/blogs/pki/firewall-rules-for-active-directory-certificate-services):
+
+- TCP/UDP port 53: DNS
+- TCP/UDP port 88: Kerberos authentication
+- TCP/UDP port 135: RPC
+- TCP/UDP port 137-138: NetBIOS
+- TCP/UDP port 389: LDAP
+- TCP/UDP port 445: SMB
+- TCP/UDP port 464: Kerberos password change
+- TCP/UDP port 636: LDAP SSL
+- TCP/UDP port 3268-3269: Global catalog
+
+AD CS additionally has the following requirements for Certificate Authorities:
+
+- TCP random port above 1023: RPC dynamic port allocation
+
+The following ports are optional depending on services used, and tend to apply to
+Certificate Enrollment Web Services:
+
+- TCP port 80: HTTP
+- TCP port 443: HTTPS
+- TCP port 445: SMB
+
+If using Active Directory Federation Services (ADFS) for single sign on the following ports are
+also required:
+
+- TCP port 80: HTTP
+- TCP port 443: HTTPS
+- TCP port 49443: ADFS
 
 ## Core Concepts
 
 TODO
 
-## Common ADCS workflows
+## Common AD CS workflows
 
 TODO
+
+## Setting up A Vulnerable AD CS Server
+### Installing AD CS
```

Comment:
Just a heads up that these checklists render poorly:

<img width="1149" alt="image" src="https://user-images.githubusercontent.com/60357436/214442879-89df3a28-7119-45bb-a109-7d0413739c7d.png">